ipfs-8-to-9/migration/go.mod

```diff
@@ -0,0 +1,21 @@
+module github.com/ipfs/fs-repo-migrations/ipfs-8-to-9/migration
```

Comment:
I am not sure there is a good way of having self-contained vendored-like dependencies  after the issues #94 . Likely net/trace needs to be kept aligned for all migrations in all submodules. Annoying but probably do-able anyways.

Reply:
> I'd manually vendor

That seems even more work.

> That, or move away from the single binary approach

I think this is the sanest option. The general binary should call the other binaries rather than importing. Also, I can imagine ipfs only downloading the migrations it needs to download rather than a big file with all.

However, we can do that outside of this PR things are not broken right now.

Reply:
> I think this is the sanest option. The general binary should call the other binaries rather than importing. Also, I can imagine ipfs only downloading the migrations it needs to download rather than a big file with all.

I agree. Even better, we can move the main "which migrations do I run" logic into go-ipfs itself.

> However, we can do that outside of this PR things are not broken right now.

That will back us into a corner where we _won't_ be able to add a new migration till we fix this. I'm not willing to take a risk like that unless we're up against a wall.

Reply:
Ok, many moving pieces. I propose:

1) Finish things here as they are, leave go-ipfs with a working tested migration-set.
2) Work on splitting migration binaries and teaching go-ipfs how to use them next. We will need to publish migration binaries separately (migrations.ipfs.io?)/

We need to avoid snowballs that end up blocking releases or leaving a lot never-merged code behind. The fact that there are 3 pending migrations is a large snowball already.

Reply:
Ok... Then let's vendor these deps for now.

Reply:
I will have to add the `vendor` folder at the end or in a different PR, since it pollutes this review.